examples/injection/module/rust/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+mod proto {
+    include!(concat!(env!("OUT_DIR"), "/oak.examples.injection.rs"));
+}
+
+use oak::{
+    grpc,
+    io::{Receiver, Sender},
+};
+use proto::{
+    blob_request::Request, BlobRequest, BlobResponse, BlobStore, BlobStoreDispatcher,
+    BlobStoreInterface, BlobStoreProviderSender, BlobStoreRequest, BlobStoreSender, GetBlobRequest,
+    PutBlobRequest,
+};
+
+oak::entrypoint!(grpc_fe => |_in_channel| {
+    oak::logger::init_default();
+    let (to_provider_write_handle, to_provider_read_handle) = oak::channel_create().unwrap();
+    let (from_provider_write_handle, from_provider_read_handle) = oak::channel_create().unwrap();
+    oak::node_create(&oak::node_config::wasm("app", "provider"), to_provider_read_handle)
+        .expect("Failed to create provider");
+    Sender::new(to_provider_write_handle)
+        .send(&BlobStoreProviderSender { sender: Some(Sender::new(from_provider_write_handle)) })
+        .expect("Failed to send handle to provider");
+
+    let frontend = BlobStoreFrontend::new(
+            Sender::new(to_provider_write_handle),
+            Receiver::new(from_provider_read_handle));
```

Comment:
It would be nice to close the unused ends of channels (`to_provider_read_handle` and `from_provider_write_handle` I think).